framework/db/ActiveQueryTrait.php

```diff
@@ -159,7 +159,10 @@ private function createModels($rows)
      */
     public function findWith($with, &$models)
     {
-        $primaryModel = new $this->modelClass;
+        $primaryModel = reset($models);
+        if (!$primaryModel instanceof ActiveRecordInterface) {
+            $primaryModel = new $this->modelClass;
```

Comment:
Ah, so avoiding is referencing the fact that in lots of cases basically you remove the need to instantiate a new object, not the fact that you use `new` when you have to instantiate an object.